docs/blog/introduction-0.4.0/index.qmd

```diff
@@ -0,0 +1,137 @@
+html-table-processing: none
+author: Rich Iannone
+date: 2024-03-19
+freeze: true
+jupyter: python3
+---
+
+The recent `v0.4.0` release of **Great Tables** contains nanoplots as a major new feature. So, in this post I'll concentrate on showing you all the things you can do with nanoplots. While the emphasis here will be on the aforementioned nanoplots feature, the last two releases (`v0.3.1` and `v0.4.0`) overall gave us a nice collection of improvements which includes:
+
+- the [`fmt_nanoplot()`](`great_tables.GT.fmt_nanoplot`) method for adding nanoplots to your table
+- improved HTML table representations in different code environments
+- integration of Polars selectors in the `columns=` arg of all formatting (`fmt_*()`) methods
+- the [`save()`](`great_tables.GT.save`) method for saving a **GT** table as an image file
+- rendering a **GT** table as an HTML string though [`as_raw_html()`](`great_tables.GT.as_raw_html`)
+
+Now let's dive into the wonderful world of nanoplots!
+
+### Nanoplots, small interactive plots in your table
+
+Nanoplots are small yet information-laden plots that fit nicely into table cells. They are interactive, allowing for more information to be shown on hovering (or through touch when that interaction is available). Nanoplots try to show individual data points with reasonably good visibility (space is limited, this is going in a table after all!) and the plot representations change depending on the data fed into them.
+
+We can generate nanoplots via the [`fmt_nanoplot()`](`great_tables.GT.fmt_nanoplot`) method. Let's make two nanoplots of the two different available plot types: `"line"` and `"bar"`:
+
+```{python}
+from great_tables import GT
+import polars as pl
+
+random_numbers_df = pl.DataFrame(
+    {
+        "i": range(1, 5),
+        "lines": [
+            "20 23 6 7 37 23 21 4 7 16",
+            "2.3 6.8 9.2 2.42 3.5 12.1 5.3 3.6 7.2 3.74",
+            "-12 -5 6 3.7 0 8 -7.4",
+            "2 0 15 7 8 10 1 24 17 13 6",
+        ],
+    }
+).with_columns(bars=pl.col("lines"))
+
+(
+    GT(random_numbers_df, rowname_col="i")
+    .fmt_nanoplot(columns="lines", plot_type="line")
+    .fmt_nanoplot(columns="bars", plot_type="bar")
+)
+```
+
+As can be seen, the method accepts bundles of values per cell that are formatted as strings (with spaces between each of the values). You can also use list columns as acceptable input. Here is an example of that, and we'll also add in reference lines and reference areas (using the `reference_line=` and `reference_area=` arguments):
+
+```{python}
+random_numbers_df = pl.DataFrame(
+    {
+        "i": range(1, 5),
+        "lines": [
+            { "val": [20.0, 23.0, 6.0, 7.0, 37.0, 23.0, 21.0, 4.0, 7.0, 16.0] },
+            { "val": [2.3, 6.8, 9.2, 2.42, 3.5, 12.1, 5.3, 3.6, 7.2, 3.74] },
+            { "val": [-12.0, -5.0, 6.0, 3.7, 0.0, 8.0, -7.4] },
+            { "val": [2.0, 0.0, 15.0, 7.0, 8.0, 10.0, 1.0, 24.0, 17.0, 13.0, 6.0] },
+        ],
+    }
```

Comment:
Having this data created above in pandas and then recreated in polars is a lot of reading. What about creating it once and using it across the two examples?

Whichever you choose, you might mention briefly how the input is being used (e.g. a string is being split into a sequence of data; or the first entry of the struct "val" is used), etc..